lib/api/objectPut.js

```diff
@@ -40,86 +41,107 @@ function _storeIt(bucketName, objectKey, objMD, authInfo, canonicalID,
     if (request.query && request.query['x-amz-acl']) {
         headers['x-amz-acl'] = request.query['x-amz-acl'];
     }
-    const metadataStoreParams = {
-        objectKey,
-        authInfo,
-        metaHeaders,
-        size,
-        contentType: request.headers['content-type'],
-        cacheControl: request.headers['cache-control'],
-        contentDisposition: request.headers['content-disposition'],
-        contentEncoding:
-            removeAWSChunked(request.headers['content-encoding']),
-        expires: request.headers.expires,
-        headers,
+
+    registry.run({
```

Comment:
Also, for puts where streaming v4 auth is used, we'll have to move where the parsing of the chunks is being done.  It's currently happening later in here https://github.com/scality/S3/blob/master/lib/services.js#L158 but could be moved so it is done before we run a predicate function.